csharp/src/Ice/ObjectAdapter.cs

```diff
@@ -75,202 +75,72 @@ public sealed class ObjectAdapter
         };
 
         private readonly Acm _acm;
+        private Task? _activateTask;
         private readonly Dictionary<CategoryPlusFacet, IObject> _categoryServantMap =
             new Dictionary<CategoryPlusFacet, IObject>();
         private readonly Dictionary<string, IObject> _defaultServantMap = new Dictionary<string, IObject>();
         private int _directCount;  // The number of direct proxies dispatching on this object adapter.
+        private TaskCompletionSource<object?>? _directDispatchCompletionSource;
+        private Task? _disposeTask;
         private readonly string _id; // adapter id
         private readonly Dictionary<IdentityPlusFacet, IObject> _identityServantMap =
             new Dictionary<IdentityPlusFacet, IObject>();
         private readonly List<IncomingConnectionFactory> _incomingConnectionFactories =
             new List<IncomingConnectionFactory>();
+        private bool _isDeactivated = false;
```

Comment:
I can't find where it's set to true.

Reply:
I removed this, and just use `_disposedTask != null`